src/VirtualPage.php

```diff
@@ -118,26 +118,26 @@ public function getHTML()
                 }
 
                 // special treatment for popup
-                if ($type == 'popup') {
+                if ($type === 'popup') {
                     $this->app->html->template->set('title', $this->app->title);
                     $this->app->html->template->setHTML('Content', parent::getHTML());
                     $this->app->html->template->appendHTML('HEAD', $this->getJS());
 
-                    $this->app->terminate($this->app->html->template->render());
+                    $this->app->terminateHTML($this->app->html->template);
```

Comment:
this is causing an issue. You should probably sending the rendering template here.
```
$this->app->terminateHTML($this->app->html->template->render());
````
see demos/modal.php on 'Virtual Page Logic' section.

Reply:
That is why I already opened issue #989, give me some some and I will fix it in this PR too :)